### PR TITLE
Add Container Patch URLS at build time

### PR DESCRIPTION
**Pull request by OrangeDrangon**
#  #

## 🗣 Description ##

Adds apply_patches.sh to keep the Dockerfile from growing too complex.
Can be used in `entrypoint.sh` as well I imagine.

Changes the optional build step to write a `status.txt` out. Later use that file to 
know whether or not foundry was installed. I was unable to
find a way to detect this built into to Docker and multistage builds do
not persist environment variables across them. I am very open to other options.
Then read back the file to determine whether or not to run the Container
Patch URLs then remove it..

Container Patch URLs must be ran in the second stage because they may
rely on the `/data` directory existing and what not. Running in the
second step guarantees the same runtime environment they normally expect.

## 💭 Motivation and Context ##

See Issue #220 

## 🧪 Testing ##

Built an image with various permutations of having and not having patch urls and credentials. Build ran successfully in all cases.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] _All_ future TODOs are captured in issues, which are referenced in code comments.
* [ ] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow project standards.
* [x] All relevant repo and/or project documentation has been updated to reflect
      the changes in this PR.
* [ ] Tests have been added to cover the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

**Comment by OrangeDrangon** [comment]
Not really sure if I did everything correctly. Sorry if there is any of the commit process wrong. I am of course open to any feedback or changes needed to get this merged.

**Comment by hugoprudente** [comment]
As a user I really enjoy this change, as I use my containers pre-built having the hability of using the patch_urls during my setup is awesome.

Can't wait to have this merged. Tested here and worked as a charm!

**Comment by hugoprudente** [comment]
Hi @OrangeDrangon if this PR is ready, can you mark it as Ready for Review so we can review it?

**Comment by OrangeDrangon** [comment]
@felddy I am not sure how to proceed with the error of `logging.sh` not being found. It works in my tests. I think the linter is confused.

File: src/apply_patches.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#! /bin/bash
+
+if [[ "${CONTAINER_PATCH_URLS:-}" ]]; then
+  source ./logging.sh
```
Comment (hugoprudente) [inline src/apply_patches.sh]: This is the bit that's making the code to failure, you don't need the the `./` .

If you check the entrypoint example `source logging.sh`

https://github.com/felddy/foundryvtt-docker/blob/develop/src/entrypoint.sh#L21

File: Dockerfile [hunk 2]
```diff
@@ -94,6 +98,14 @@ EXPOSE 30000/TCP
 # EXPOSE 33478/UDP
 # EXPOSE 49152-65535/UDP
 
+COPY --from=optional-release-stage /root/status.txt .
+
+RUN \
+  if  [[ $(cat status.txt) = "installed" ]]; then \
+    source ./apply_patches.sh; \
```
Comment (hugoprudente) [inline Dockerfile:105]: We may want to remove the `./` from here too, just to be consistent.

**Review by hugoprudente — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (hugoprudente) [reply Dockerfile:105]: Please also rebase your change as it's outdated with target branch as per `This branch is out-of-date with the base branch`

Comment (OrangeDrangon) [reply Dockerfile:105]: Will do sometime soon.

**Review by hugoprudente — APPROVED** [review]

**Pull request closed without merging**